In [1]:
import pandas as pd
import numpy as np
import statsmodels.formula.api as smf

In [3]:
df = pd.read_csv("gradedata2.csv")
df.head()

fname     lname  gender  age  exercise  hours  grade  \
0   Marcia      Pugh  female   17         3     10   82.4   
1   Kadeem  Morrison    male   18         4      4   78.2   
2     Nash    Powell    male   18         5      9   79.3   
3  Noelani    Wagner  female   14         2      7   83.2   
4  Noelani    Cherry  female   18         4     15   87.4   

                                      address  
0        7379 Highland Rd. , Dublin, GA 31021  
1          8 Bayport St. , Honolulu, HI 96815  
2           Encino, CA 91316, 3 Lilac Street   
3  Riverview, FL 33569, 9998 North Smith Dr.   
4   97 SE. Ocean Street , Bethlehem, PA 18015

In [4]:
df["gender"] = np.where(df['age'] == 18, 'unknown', df['gender'])

df.head(25)

fname      lname   gender  age  exercise  hours  grade  \
0    Marcia       Pugh   female   17         3     10   82.4   
1    Kadeem   Morrison  unknown   18         4      4   78.2   
2      Nash     Powell  unknown   18         5      9   79.3   
3   Noelani     Wagner   female   14         2      7   83.2   
4   Noelani     Cherry  unknown   18         4     15   87.4   
5      Neil    Whitley     male   16         5     16   88.7   
6     Nelle     Golden   female   17         1      9   80.2   
7   Armando    Hoffman     male   17         5     18   95.1   
8   Illiana      Rojas   female   15         5      9   76.5   
9      Neil     Wooten     male   15         3     15   89.7   
10   Daquan    Alvarez     male   16         2     13   85.2   
11     Nola  Velazquez   female   15         2     10   75.3   
12    Quinn     Warren   female   14         4     12   80.7   
13  Frances  Velasquez   female   15         2     15   84.2   
14  Lareina      Poole  unknown   18         1     14   87.6   
15    Medge   Mccarthy   female   15         1      8   75.8   
16     Kibo      Gates     male   16         1     10   88.2   
17    Libby     Guzman   female   19         1     19  100.0   
18   Shelly    Rosario  unknown   18         4     13   84.3   
19     Lane       Tate     male   19         4     11   84.2   
20  Isadora       Case  unknown   18         3     11   79.1   
21    Maggy  Whitfield   female   15         1     15   90.5   
22    Elton     Wagner     male   16         2      9   71.0   
23    Lance   Benjamin     male   14         5     18   90.3   
24     Kyle    Skinner     male   17         5      6   82.4   

                                              address  
0                7379 Highland Rd. , Dublin, GA 31021  
1                  8 Bayport St. , Honolulu, HI 96815  
2                   Encino, CA 91316, 3 Lilac Street   
3          Riverview, FL 33569, 9998 North Smith Dr.   
4           97 SE. Ocean Street , Bethlehem, PA 18015  
5          8041 Sunbeam Dr. , Downers Grove, IL 60515  
6           8701 Rockville Ave. , Grovetown, GA 30813  
7   9 Old Tarkiln Hill Court , Council Bluffs, IA ...  
8           133 Valley Farms Lane , Ephrata, PA 17522  
9                   43 Wood Court , Lincoln, NE 68506  
10              6 Laurel St. , Ormond Beach, FL 32174  
11          Hopkinsville, KY 42240, 495 Cypress Road   
12           Reynoldsburg, OH 43068, 81 Somerset Dr.   
13                  Roswell, GA 30075, 41 Branch St.   
14    Salt Lake City, UT 84119, 515 Wintergreen Ave.   
15           8985 Bishop Lane , Painesville, OH 44077  
16                     9855 Fawn Lane , Roy, UT 84067  
17               92 Williams Ave. , Burbank, IL 60459  
18               54 Crescent Drive , Groton, CT 06340  
19              557 Rockwell Street , Irwin, PA 15642  
20                  Milton, MA 02186, 2 Third Street   
21                 99 Sage Street , Bozeman, MT 59715  
22     Powder Springs, GA 30127, 81 East Logan Drive   
23                24 Cardinal St. , Trenton, NJ 08610  
24                Lawndale, CA 90260, 860 Henry Ave.

In [5]:
result = smf.ols('grade ~ age + exercise + hours + C(gender)', data=df).fit()

result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  grade   R-squared:                       0.665
Model:                            OLS   Adj. R-squared:                  0.664
Method:                 Least Squares   F-statistic:                     790.3
Date:                Thu, 19 Nov 2020   Prob (F-statistic):               0.00
Time:                        13:00:33   Log-Likelihood:                -6298.9
No. Observations:                2000   AIC:                         1.261e+04
Df Residuals:                    1994   BIC:                         1.264e+04
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
========================================================================================
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
Intercept               57.8721      1.401     41.317      0.000      55.125      60.619
C(gender)[T.male]       -0.5008      0.278     -1.798      0.072      -1.047       0.045
C(gender)[T.unknown]    -0.4313      0.388     -1.111      0.267      -1.193       0.330
age                      0.0571      0.081      0.707      0.480      -0.101       0.216
exercise                 0.9832      0.089     11.059      0.000       0.809       1.158
hours                    1.9175      0.031     61.608      0.000       1.856       1.979
==============================================================================
Omnibus:                      328.226   Durbin-Watson:                   2.047
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             2326.179
Skew:                          -0.573   Prob(JB):                         0.00
Kurtosis:                       8.158   Cond. No.                         226.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

y = b + (m1 * x1) + (m2 * x2) + (m3 * x3) + (m4 * x4) + (m5 * x5)

grade = 57.8721 + (genderMALE * x1) + (genderUNKNOWN * x2) + (AGE * x3) + (EXERCISE * x4) + (HOURS * x5)

grade = 57.8721 + (-0.5008 * x1) + (-0.4313 * x2) + (0.0571 * x3) + (0.9832 * x4) + (1.9175 * x5)

### The student Gender is unknown, they are 18, they exercise 5 hours per week, and study 10 hours per week

grade = 57.8721 + (-0.5008 * 0) + (-0.4313 * 1) + (0.0571 * 18) + (0.9832 * 5) + (1.9175 * 10)